californium-core/src/main/java/org/eclipse/californium/core/CoapClient.java

```diff
@@ -322,6 +315,19 @@ public CoapClient setExecutor(ExecutorService executor, boolean detach) {
 		return this;
 	}
 
+	private ScheduledThreadPoolExecutor getSecondaryExecutor() {
+		if (secondaryExecutor == null) {
+			synchronized (this) {
+				if (secondaryExecutor == null) {
+					secondaryExecutor = new ScheduledThreadPoolExecutor(1,
+							new NamedThreadFactory("CoapClient(secondary)#"));
+				}
+				this.detachExecutor = false;
```

Comment:
If a executor and a secondaryExecutor == null is used, the this may overwrite the set `detach`.

Reply:
With https://github.com/eclipse/californium/pull/937#issuecomment-493077130 this should not be an issue anymore.